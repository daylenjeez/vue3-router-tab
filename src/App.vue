<script lang="ts">
import { useRouter } from "vue-router";
import { useRouterTab } from "./plugins/vue3-router-tab";
import "./style/index.less";



export default {
  setup() {
    const router = useRouter();
    const routerTab = useRouterTab();

    return {
      handlePush(value: string) {
        router.push(value);
      },
    };
  },
};
</script>

<template>
  <div
    class
    style="margin-bottom: 30px"
  >
    <button @click="() => handlePush('/home')">
      首页
    </button>
    <button @click="() => handlePush('/list?id=1')">
      用户列表+query
    </button>
    <button @click="() => handlePush('/list?id=2')">
      用户列表+query
    </button>
    <button @click="() => handlePush('/detail/2')">
      用户详情+params2
    </button>
    <button @click="() => handlePush('/detail/3')">
      用户详情+params3
    </button>
    <router-tab />
  </div>
</template>
