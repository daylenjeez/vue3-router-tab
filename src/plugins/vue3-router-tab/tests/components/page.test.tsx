import { describe, expect, it } from 'vitest';
import Page from '../../components/page/index.vue';
import App from "../../../../App.vue";
import { getRouterTab, reset,router } from '../common';
import { mount } from '@vue/test-utils';

const _routerTab = getRouterTab();
const _reset = reset.bind(null, _routerTab);

describe('YourComponent', () => {
  
  it('renders the correct route component', async () => {
    _reset();
    const wrapper = mount(App, {global: {plugins: [router]}});

    await router.push('/initial');

    await wrapper.vm.$nextTick();
    
    expect(wrapper.findComponent(Page).exists());
  });
});
