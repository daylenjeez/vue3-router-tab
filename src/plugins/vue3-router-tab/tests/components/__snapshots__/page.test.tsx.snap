// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueRouterTab Plugin > should render the page 1`] = `
"<div>
  <!---->
</div>"
`;
